### PR TITLE
feat: add overload method

### LINKED ISSUE #1495: Add overload method with `ResourceName` as a formal parameter
This is part of the project of Hermetic Generation for Client Libraries.

In order to maintain backward compatibility, some libraries, e.g, `java-monitoring` need an overload method with with `ResourceName` as its formal parameter.

### PR DESCRIPTION
**Pull request by JoeWang1127**
Thank you for opening a Pull Request! For general contributing guidelines, please refer to [contributing guide](https://github.com/googleapis/gapic-generator-java/blob/main/CONTRIBUTING.md)

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/gapic-generator-java/issues/new/choose) before writing your code! That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #1495 ☕️

### CODE REVIEW

**Review by suztomo — COMMENTED** [review]
The term “customization” is obscure.

**Comment by suztomo** [comment]
Can you update the title of this pull request more specific?

File: gapic-generator-java/src/main/java/com/google/api/generator/gapic/utils/ParserUtils.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Optional;
+import org.yaml.snakeyaml.LoaderOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.SafeConstructor;
+
+public class ParserUtils {
```
Comment (suztomo) [inline gapic-generator-java/src/main/java/com/google/api/generator/gapic/utils/ParserUtils.java:15]:
```suggestion
public class ConfigurationUtils {
```

**Review by suztomo — COMMENTED** [review]
Where is the code that replaced FolderName with ResourceName?

**Comment by JoeWang1127** [comment]
> Where is the code that replaced FolderName with ResourceName?

I haven't added the logic yet.

I'm not sure whether this is a good approach, doing research now.

**Comment by JoeWang1127** [comment]
Closed as it might not be needed.

**Pull request closed without merging**